compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -473,4 +497,34 @@ private sealed class GeneratedMethod {
   class BindingMethod(override val spec: FunSpec) : GeneratedMethod()
 }
 
+private data class Binding(
+  val contributedClass: ClassDescriptor,
+  val annotation: AnnotationDescriptor,
+  val boundType: ClassDescriptor
+)
+
+private fun List<Binding>.findHighestPriorityBinding(): Binding {
+  if (size == 1) return this[0]
+
+  val bindingsByPriority = groupBy { it.annotation.priority() }
+    .filter { it.value.isNotEmpty() }
+
+  val bindings = bindingsByPriority[HIGHEST]
```

Comment:
You could've sort `bindingsByPriority` and take `.firstOrNull { it.isNotEmpty() }` instead of hardcoding priorities.

Reply:
I considered this first, but didn't do it because it's a map. But I just learnt that Kotlin has `SortedMap` and sorting a map with max 3 items is very quick. I updated that.